CorpusSearch/ClientApp/src/routes/Newbies.tsx

```diff
@@ -0,0 +1,45 @@
+import React, {useEffect, useState} from "react"
+function CorpusLink({ident="",name=""}): JSX.Element {
+    if ( name.slice(0,2) != "🎥"){
+        name = "📖" + name
+    }
+    return <div>
+        <a href={"docs/" + ident}>{name}</a>
+    </div>
+}
+export const Newbies = () => {
+type DocType =  {name:string, ident:string, uploaded:string}
+const [newDocs, setNewDocs  ] 
+    = useState<DocType[]>([])
+useEffect(()=>  {
+    const fetchNewDocs = async () => {
+        await fetch("api/metadata/latest")
+            .then((res)=> {
+                if (!res.ok) {
+                    console.warn(`metadata/latest returned status: ${res.status}`)
+                }
+                return res.json()
+            }).then((json: DocType[])=> {
+                setNewDocs(json)
+        })
```

Comment:
Sure, but would have to get the casting to DocType[] correct.

Reply:
You should be able to use `as DocType[]`

Reply:
Will try. Otherwise TS2345: Argument of type 'Promise<any>' is not assignable to parameter of type 'SetStateAction<DocType[]>'.

Reply:
Doesn't that too much either:
<html>TS2352: Conversion of type 'Promise&lt;any&gt;' to type 'DocType[]' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.<br/>Type 'Promise&lt;any&gt;' is missing the following properties from type 'DocType[]': length, pop, push, concat, and 29 more.